packages/web/src/brandkit/common/Page.tsx

```diff
@@ -0,0 +1,242 @@
+
+    sections: [
+      { title: 'Overview', href: `${LOGO_PATH}#${hashNav.brandLogo.overview}` },
+      { title: 'Space and Sizing', href: `${LOGO_PATH}#${hashNav.brandLogo.space}` },
+      { title: 'Backgrounds', href: `${LOGO_PATH}#${hashNav.brandLogo.backgrounds}` },
+    ],
+  },
+  {
+    title: 'Color',
+    href: COLOR_PATH,
+
+    sections: [
+      { title: 'Overview', href: `${COLOR_PATH}#${hashNav.brandColor.overview}` },
+      { title: 'Background Colors', href: `${COLOR_PATH}#${hashNav.brandColor.backgrounds}` },
+    ],
+  },
+  {
+    title: 'Typography',
+    href: TYPE_PATH,
+
+    sections: [
+      { title: 'Overview', href: `${TYPE_PATH}#${hashNav.brandTypography.overview}` },
+      { title: 'Type Scale', href: `${TYPE_PATH}#${hashNav.brandTypography.scale}` },
+    ],
+  },
+  {
+    title: 'Icons',
+    href: ICONS_PATH,
+    sections: [],
+  },
+  // {
+  //   title: 'Key Imagery',
+  //   href: IMAGERY_PATH,
+
+  //   sections: [
+  //     { title: 'Overview', href: `${IMAGERY_PATH}#${hashNav.brandImagery.overview}` },
+  //     { title: 'Illustrations', href: `${IMAGERY_PATH}#${hashNav.brandImagery.illustrations}` },
+  //     { title: 'Abstract Graphics', href: `${IMAGERY_PATH}#${hashNav.brandImagery.graphics}` },
+  //   ],
+  // },
+]
+
+export const ROUTE_TO_TITLE = PAGES.reduce((mapping, page) => {
+  mapping[page.href] = page.title
+  return mapping
+}, {})
+
+interface Section {
+  id: string
+  children: React.ReactNode
+}
+
+interface Props {
+  router: SingletonRouter
+  sections: Section[]
+}
+
+interface State {
+  routeHash: string
+}
+
+class Page extends React.Component<Props & ScreenProps, State> {
+  state: State = {
+    routeHash: '',
+  }
+
+  ratios: Record<string, { id: string; ratio: number; top: number }> = {}
+
+  observer: IntersectionObserver
+
+  pageRef = React.createRef<View>()
+
+  sectionRefs = this.props.sections.reduce((acc, section) => {
+    acc[section.id] = React.createRef<View>()
+    return acc
+  }, {})
+
+  onChangeHash = () => {
+    this.setState({ routeHash: window.location.hash })
+  }
+
+  onIntersection = (entries: IntersectionObserverEntry[]) => {
+    this.ratios = entries
+      .map((entry) => ({
+        id: entry.target.id,
+        ratio: entry.intersectionRatio,
+        top: entry.boundingClientRect.top,
+      }))
+      .reduce((acc, currentValue) => {
+        acc[currentValue.id] = currentValue
+        return acc
+      }, this.ratios)
+
+    const top = Object.keys(this.ratios)
+      .map((key) => this.ratios[key])
+      .reduce(
+        (acc, current) => {
+          if (current.ratio > acc.ratio) {
+            return current
+          }
+          return acc
+        },
+        { ratio: 0, id: this.state.routeHash }
+      )
+
+    if (this.state.routeHash !== top.id) {
+      this.setState({ routeHash: top.id })
+      window.history.replaceState({}, top.id, `${location.pathname}#${top.id}`)
+    }
+  }
+
+  observation = () => {
```

Comment:
same here - observeHeaderScroll, or something like that so that i know the purpose of this is to update the hash